### PR TITLE
add support for workspaceid based filter

### PR DESCRIPTION
**Pull request by sudhanshutech**
**Notes for Reviewers**

- This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by sudhanshutech** [comment]
@Yashsharma1911 please give your feedback and what's missing .

File: server/internal/graphql/model/models_gen.go [hunk 1]
```diff
@@ -82,6 +82,7 @@ type CatalogSelector struct {
 	PatternType []*string `json:"patternType,omitempty"`
 	Userid      []*string `json:"userid,omitempty"`
 	OrgID       []*string `json:"orgID,omitempty"`
+	WorkspaceID []*string `json:"workspaceID,omitempty"`
```
Comment (amitamrutiya) [inline server/internal/graphql/model/models_gen.go:85]: Have you written this manually this is an autogenerated file from graphql. Run make graphql after change in the schema

**Review by amitamrutiya — COMMENTED** [review]
generated.go file has to autogenerated based on the schema changes, that is missing.

File: server/models/remote_provider.go [hunk 2]
```diff
@@ -1810,6 +1810,12 @@ func (l *RemoteProvider) GetCatalogMesheryPatterns(tokenString string, page, pag
 			q.Add("userid", user)
 		}
 	}
+
+	if len(workspaceID) > 0 {
+		for _, workspace := range workspaceID {
+			q.Add("workspaceID", workspace)
+		}
+	}
```
Comment (Yashsharma1911) [inline server/models/remote_provider.go:1818]: We don't need to pass multiple workspace ids in request as remote provider only accept single workspace id at a time

File: server/internal/graphql/resolver/catalog_content.go [hunk 3]
```diff
@@ -60,13 +60,20 @@ func (r *queryResolver) fetchCatalogPattern(ctx context.Context, provider models
 		}
 	}
 
+	workspaceID := []string{}
+	for _, workspaceIDPtr := range selector.WorkspaceID {
+		if workspaceIDPtr != nil {
+			workspaceID = append(workspaceID, *workspaceIDPtr)
+		}
+	}
```
Comment (Yashsharma1911) [inline server/internal/graphql/resolver/catalog_content.go:68]: Same here remove this line as you don't need it

File: server/models/remote_provider.go [hunk 4]
```diff
@@ -1810,6 +1810,12 @@ func (l *RemoteProvider) GetCatalogMesheryPatterns(tokenString string, page, pag
 			q.Add("userid", user)
 		}
 	}
+
+	if len(workspaceID) > 0 {
+		for _, workspace := range workspaceID {
+			q.Add("workspaceID", workspace)
```
Comment (amitamrutiya) [inline server/models/remote_provider.go:1816]:
```suggestion
			q.Add("workspaceid", workspace)
```

**Comment by Yashsharma1911** [comment]
@sudhanshutech although we merged please do consider my above comments to avoid regressions

**Comment by sudhanshutech** [comment]
> @sudhanshutech although we merged please do consider my above comments to avoid regressions

yeah i have that in mind. We merged since it was required but we do require multiple support since only then autocomplete component make sense.

**Pull request merged**